### PR TITLE
`cudf::ast` Small Refactorings

### PR DESCRIPTION
**Pull request by codereport**
While reading/debugging through AST code in order to understand it, I did/am doing a few small refactorings.

### CODE REVIEW

**Review by rgsl888prabhu — APPROVED** [review]

File: cpp/src/ast/transform.cu [hunk 1]
```diff
@@ -138,24 +136,23 @@ std::unique_ptr<column> compute_column(table_view const table,
     cudf::mutable_column_device_view::create(output_column->mutable_view(), stream);
 
   // Configure kernel parameters
-  auto const num_intermediates     = expr_linearizer.get_intermediate_count();
-  auto const shmem_size_per_thread = static_cast<int>(sizeof(std::int64_t) * num_intermediates);
+  auto const num_intermediates = expr_linearizer.intermediate_count();
+  // size_per_thread/block, limit_per_block are for shmem (shared memory)
+  auto const size_per_thread = static_cast<int>(sizeof(std::int64_t) * num_intermediates);
```
Comment (jrhemstad) [inline cpp/src/ast/transform.cu:141]: I think the `shmem` prefix was actually helpful here.

Comment (codereport) [reply cpp/src/ast/transform.cu:141]: ok, i will add it back.

**Review by jrhemstad — APPROVED** [review]

**Comment by codereport** [comment]
@gpucibot merge

**Comment by codereport** [comment]
rerun tests

**Pull request merged**